### PR TITLE
Practice labs

### PR DESCRIPTION
**Pull request by HawkeyeUW**
Submit practice labs

### CODE REVIEW

File: students/jvirtue/lesson04/mailroom_pt2.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        print("4: Quit")
+        response = input("Please select option [Enter number 1 - 4]: ")
+        if response.title() == "1":
+            send_thanks(donors_db)
+        elif response.title() == "2":
+            create_report(donors_db)
+        elif response.title() == "3":
+            thank_you_letter(donors_db)    
+        elif response.title() == "4":
+            break
```
Comment (natasha-aleksandrova) [inline students/jvirtue/lesson04/mailroom_pt2.py]: per assignment instructions you should be using a switch dict here

File: students/jvirtue/lesson04/mailroom_pt2.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    sort_table = crunch_numbers(db_name)
+    for item in sort_table:
+        s = f'{item[0]:<20}{" ":<1}{"$":<1}{item[1]:>14}{" ":<1}{item[2]:>15}{" ":<1}{"$":<1}{item[3]:>14}'
+        print(s)
+
+
+#Options Menu
+if __name__ == '__main__':
+    #donors_db = [["Fred Flintstone", 100,200],["Wilma Flintstone", 300],["Bamm-Bamm Rubble", 50,30,40],["Barney Rubble", 75],["Pebbles Flintstone",500]]
+    donors_db = {'Fred Flintstone': [100,200],'Wilma Flintstone': [300],'Bamm-Bamm Rubble': [50,30,40],'Barney Rubble' : [75],'Pebbles Flintstone': [500]}
```
Comment (natasha-aleksandrova) [inline students/jvirtue/lesson04/mailroom_pt2.py]: 👍

File: students/jvirtue/lesson04/mailroom_pt2.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+def crunch_numbers(db_name):
+    crunch_numbers = []
+    for i in db_name:
+        donor = i[0]
+        donations = i[1:]
+        sum_gift = sum(donations)
+        count = len(donations)
+        avg_gift = sum_gift / count
+        crunch_numbers.append([donor,sum_gift,count,avg_gift])
+        crunch_numbers.sort(key=sort_key,reverse=True)
```
Comment (natasha-aleksandrova) [inline students/jvirtue/lesson04/mailroom_pt2.py]: you don't want to sort this in a loop.. that's wasteful

File: students/jvirtue/lesson04/mailroom_pt2.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    return db_name.setdefault(name,[]).append(amt)
+
+def thank_you(person,amount):
+    return "Thank you {} ".format(person) + "for the donation in the amount of {}!".format(amount)
+
+def thank_you_letter(db_name):
+    return f"""Dear {db_name["name"]},\nThank you for your very generous donation of ${db_name["amount"]:.2f}.\n{"Sincerely":>40}\n{"Fred Flintstone":>50}"""
+
+#Create a report
+def crunch_numbers(db_name):
```
Comment (natasha-aleksandrova) [inline students/jvirtue/lesson04/mailroom_pt2.py]: what is `db_name` ? isn't that just your `donors_db`? why not use it directly?

**Comment by HawkeyeUW** [comment]
Natasha,

I did finish and submit Mailroom pt2, yesterday.  It was part of pull request #1439.  This one is complete : )  I didn’t see any update on learning site.

Jason

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by natasha-aleksandrova** [comment]
Thanks for letting me know, I corrected it.

We get a lot of submissions and don't always have context on partial submissions so I was a bit confused on the mailroom in the PR #1439. It helps if you include comments in the PR and link to another PR and give a bit more context. Thanks!

**Comment by HawkeyeUW** [comment]
I will do this in the future.  Thanks again.

Jason

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Pull request merged**